lib/IRGen/EnumPayload.cpp

```diff
@@ -564,6 +564,9 @@ EnumPayload::emitCompare(IRGenFunction &IGF, APInt mask, APInt value) const {
       v = IGF.Builder.CreateBitOrPointerCast(v, intTy);
     
     if (isMasked) {
+#if defined(__BIG_ENDIAN__) && defined(__LP64__)
+      v  = IGF.Builder.CreateLShr(v, 56);
+#endif
```

Comment:
This needs to be gated on the target triple, not a preprocessor conditional.

Reply:
Hi @compnerd,  I am following the similar style of some of the previous commits, e.g., https://github.com/apple/swift/commit/9d0730d876ef85d4727e10132af08a94c0fe3661. What do you suggest to do specifically?

Reply:
Yes, I would create member variables in IRGenModule that are initialized in its constructor so that you can write:
```
if (IGM.IsBigEndian && IGM.Is64Bit)
```

Reply:
I would suggest using 
```
if ( !IGM.Triple.isLittleEndian() && IGM.Triple.isArch64Bit())
```
since `Triple` is already a member variable defined handy in `IRGenModule`.